crates/consensus/src/transaction/mod.rs

```diff
@@ -59,6 +103,17 @@ pub trait Transaction: any::Any + Send + Sync + 'static {
 
     /// Get `data`.
     fn input(&self) -> &[u8];
+
+    /// Returns the transaction type
+    fn ty(&self) -> u8;
+
+    /// Returns the EIP2930 `access_list` for the particular transaction type. Returns `None` for
+    /// older transaction types.
+    fn access_list(&self) -> Option<&AccessList>;
+
+    /// Blob versioned hashes for eip4844 transaction. For previous transaction types this is
+    /// `None`.
+    fn blob_versioned_hashes(&self) -> Option<Vec<&B256>>;
```

Comment:
blob versioned hashes is a part of the tx and should not require an allocation
```suggestion
    fn blob_versioned_hashes(&self) -> Option<&[B256]>;
```